src/app/pages/course-details/course-details.component.ts

```diff
@@ -7,16 +12,52 @@ import { Component, ViewEncapsulation, OnInit, OnDestroy, ChangeDetectionStrateg
 	template: require('./course-details.template.html'),
 	changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class CourseDetailsPageComponent implements OnInit, OnDestroy {
-	constructor() {
+export class CourseDetailsPageComponent extends BasePage {
+
+	private courseId: string;
+	private course: ICourse;
+
+	constructor(
+		private ref: ChangeDetectorRef,
+		private route: ActivatedRoute,
+		private courseService: CourseService,
+		private loaderService: LoaderService
+	) {
+		super();
+	}
 
+	onInit() {
+		this.subscribeCourseId();
 	}
 
-	public ngOnInit() {
+	onDestroy() {
 
 	}
 
-	public ngOnDestroy() {
+	private subscribeCourseId() {
+		let sub = this.route.params.subscribe((params) => {
```

Comment:
mozna uzyc operatora 'pick'

this.routex
.params
.pick('id')
.subscribe((id) => {
this.courseId = id;
this.subscribeCourse();
}